odxtools/nameditemlist.py

```diff
@@ -20,7 +20,7 @@ def short_name(self) -> str:
 TNamed = TypeVar("TNamed", bound=OdxNamed)
 
 
-class ItemAttributeList(Generic[T]):
+class ItemAttributeList(List[T]):
```

Comment:
`typing.List` is a deprecated alias to `list`, and `list` is not supposed to be extended by user, UserList should be used as base class instead.

Also, the property _item_list makes no sense if the ItemAttributeList is already a list

See https://docs.python.org/3/library/collections.html#collections.UserList

Reply:
Consider using https://docs.python.org/3/library/collections.abc.html#collections.abc.Collection instead

Reply:
hm, unfortunatly deriving from `UserList` does not seem to work as intended: letting mypy loose on
```python
from odxtools.nameditemlist import NamedItemList
from typing import List

class Item:
    def __init__(self, sn: str) -> None:
        self.short_name = sn

def foo(x: List[Item]) -> None:
    print(len(x))

nil = NamedItemList([Item("a"), Item("b")])

foo(nil)
```

produces
```
bla.py:13: error: Argument 1 to "foo" has incompatible type "NamedItemList[Item]"; expected "list[Item]"  [arg-type]
Found 1 error in 1 file (checked 1 source file)
```

any ideas?

Reply:
try https://docs.python.org/3/library/collections.abc.html#collections.abc.Sequence interface instead

Reply:
the problem with that is that all types would need to be annotated with `Sequence` instead of `List`. I find that rather unintuitive. In the [latest version](https://github.com/mercedes-benz/odxtools/pull/251/commits/67b321891543f359c2ed1090064c7826d447cbdf), I've implemented all public methods of `list`. I'm aware that it's not perfect, but I think it is more than good enough for the intended use case of `NamedItemList`.

Reply:
list is not an interface, it's a class, you either inherit from it or you don't.

From purely object oriented point of view NamedItemList will not be a List regardless of what functions you implement.

One example where what you are doing will break, is when python 3.13 decides to add new function to list class.

Another one is isinstance check.

Reply:
> list is not an interface, it's a class, you either inherit from it or you don't.

after this PR we do ;)

for now we have to inherit from capital-L list, though because `list` does not yet support generics in python 3.8. (alternatively we can raise the minimum python version to 3.10 and derive from lowercase-l list.)

> From purely object oriented point of view NamedItemList will not be a List regardless of what functions you implement.

but if we inherit from it, it is a list, right?

> One example where what you are doing will break, is when python 3.13 decides to add new function to list class.

as long as this new function would have no effect on the attributes exposed, it would work just fine. This is already the case with some methods, e.g., `sort()` or `reverse()` are not overloaded by `NamedItemList` with this patch.

> Another one is isinstance check.

that is basically the point of this PR: I want `isinstance(named_item_list, list)` to be `True` to drive home the point that this class *is* a list
```
>>> isinstance(ecu.services, list)
True
```
while without it, we get `False`...